src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberContainerSymbol.cs

```diff
@@ -2326,26 +2289,105 @@ internal Binder GetBinder(CSharpSyntaxNode syntaxNode)
             return this.DeclaringCompilation.GetBinder(syntaxNode);
         }
 
-        private static void MergePartialMethods(
+        private static void MergePartialAndReplacedMembers(
             Dictionary<string, ImmutableArray<Symbol>> membersByName,
             DiagnosticBag diagnostics)
         {
+            var membersGroupedBySignature = new Dictionary<Symbol, OneOrMany<Symbol>>(MemberSignatureComparer.DuplicateSourceComparer);
             //key and value will be the same object
-            var methodsBySignature = new Dictionary<MethodSymbol, SourceMethodSymbol>(MemberSignatureComparer.DuplicateSourceComparer);
+            var partialMethodsBySignature = new Dictionary<MethodSymbol, SourceMethodSymbol>(MemberSignatureComparer.DuplicateSourceComparer);
 
             foreach (var name in membersByName.Keys.ToArray())
             {
-                methodsBySignature.Clear();
-                foreach (var symbol in membersByName[name])
+                var members = membersByName[name];
+                if (members.Length < 2)
                 {
-                    var method = symbol as SourceMethodSymbol;
+                    continue;
+                }
+
+                membersGroupedBySignature.Clear();
+
+                foreach (var member in members)
+                {
+                    switch (member.Kind)
+                    {
+                        case SymbolKind.Method:
+                        case SymbolKind.Property:
+                        case SymbolKind.Event:
+                            OneOrMany<Symbol> group;
+                            if (membersGroupedBySignature.TryGetValue(member, out group))
+                            {
+                                membersGroupedBySignature[member] = group.Add(member);
+                            }
+                            else
+                            {
+                                membersGroupedBySignature.Add(member, OneOrMany.Create(member));
+                            }
+                            break;
+                    }
+                }
+
+                foreach (var group in membersGroupedBySignature.Values)
+                {
+                    if (group.Count < 2)
+                    {
+                        continue;
+                    }
+
+                    Symbol last = null;
+                    foreach (var member in group)
+                    {
+                        var method = member as SourceMethodSymbol;
+                        if ((object)method != null && method.IsPartial)
+                        {
+                            continue; // PROTOTYPE: Allow replacing partial methods.
```

Comment:
Even though `PROTOTYPE` comments should be removed before merging into the future branch?